screeps-start.js

```diff
@@ -81,6 +91,74 @@ const installPackages = () => {
   }
 
   console.log("Done updating");
+}
+
+const updatePackages = (doUpdate) => {
+  const mods = config.mods || [];
+  const bots = config.bots || {};
+
+  const modsPackage = loadPackage(ModsDir);
+  const dependencies = modsPackage.dependencies || {};
+
+  // Calculate package diff
+  const configuredPackages = [...mods, ...Object.values(bots)];
+
+  const packagedMods = configuredPackages.filter(
+    (pkg) =>
+      Object.entries(dependencies).some((dependency) =>
+        isDependency(pkg, dependency),
+      ),
+  ).map((pkg) => parseVersionSpec(pkg));
+
+  let outdated = {};
+  const outdatedFile = path.resolve(ModsDir, "outdated.json");
+  try {
+    // `npm outdated --json` returns 1 if there are outdated packages,
+    // which causes `execSync` to throw an error.
+    execSync("npm outdated --json > outdated.json || true", {
+      cwd: ModsDir,
+      stdio: "inherit",
+      encoding: "utf8",
+      shell: true,
+    })
+    const output = fs.readFileSync(outdatedFile)
+    outdated = JSON.parse(output);
+  } catch {
```

Comment:
Do we actually need the empty catch?
```suggestion
```
Can't remember if it's only newer versions that allow omitting catch..